internal/tba/tba.go

```diff
@@ -0,0 +1,131 @@
+package tba
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/Pigmice2733/peregrine-backend/internal/store"
+)
+
+// Service provides an interface to retreive data from
+// The Blue Alliance's API
+type Service struct {
+	URL    string
+	APIKey string
+}
+
+type tbaDistrict struct {
+	Abbreviation string `json:"abbreviation"`
+}
+
+type tbaWebcast struct {
+	Type    string `json:"type"`
+	Channel string `json:"channel"`
+}
+
+type tbaEvent struct {
+	Key          string       `json:"key"`
+	ShortName    string       `json:"short_name"`
+	District     *tbaDistrict `json:"district"`
+	Lat          float64      `json:"lat"`
+	Lng          float64      `json:"lng"`
+	LocationName string       `json:"location_name"`
+	Week         *int         `json:"week"`
+	StartDate    string       `json:"start_date"`
+	EndDate      string       `json:"end_date"`
+	TimeZone     string       `json:"timezone"`
+	Webcasts     []tbaWebcast `json:"webcasts"`
+}
+
+// Maximum size of response from the TBA API to read. This value is about 4x the
+// size of a typical /events/{year} response from TBA.
+const maxResponseSize int64 = 1.2e+6
+
+func (s *Service) makeRequest(path string) (*http.Response, error) {
+	req, err := http.NewRequest("GET", s.URL+path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("X-TBA-Auth-Key", s.APIKey)
+
+	return http.DefaultClient.Do(req)
```

Comment:
We should probably not sure the DefaultClient. https://medium.com/@nate510/don-t-use-go-s-default-http-client-4804cb19f779